### PR TITLE
Make JwtTokenHelper non-static, add IJwtGenerator interface and add as Singleton Service (also refactor code)

### LINKED ISSUE #4: Check if JwtHelper Class should be made singleton instead of static
Never seen a "static service" before lol

### PR DESCRIPTION
**Pull request by baltermia**

### CODE REVIEW

File: Core/WebAPI/Controllers/AuthController.cs [hunk 1]
```diff
@@ -10,28 +10,21 @@ namespace Humteria.WebAPI.Controllers;
 [ApiController]
 public class AuthController : ControllerBase
 {
-    private readonly IConfigurationRoot _configuration;
-    private readonly IMapper _mapper;
-    private readonly IMainInterface _repository;
+    private readonly IMapper m_mapper;
```
Comment (mfhum) [inline Core/WebAPI/Controllers/AuthController.cs]: why m_ and not just _ ?

File: Core/WebAPI/Controllers/AuthController.cs [hunk 2]
```diff
@@ -10,28 +10,21 @@ namespace Humteria.WebAPI.Controllers;
...
-            .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
-            .Build();
-        _mapper = mapper;
-        _repository = repository;
-
-        if (_configuration == null || _mapper == null || _repository == null)
-        {
-            throw new ArgumentNullException();
-        }
+        m_jwtGenerator = jwtGenerator;
```
Comment (mfhum) [inline Core/WebAPI/Controllers/AuthController.cs]: same here

File: Core/WebAPI/Helpers/JwtTokenHelper.cs [hunk 3]
```diff
@@ -5,83 +5,77 @@
...
-    private static readonly string _TokenSecret;
-    private static readonly string _Audience;
-    private static readonly string _Issuer;
+    public string GenerateToken(JWTUserForTokenDTO user, int daysToExpire = 5);
+    public Guid ValidateToken(string token);
+}
 
+public class JwtTokenHelper : IJwtGenerator
+{
+    private readonly string m_tokenSecret;
```
Comment (mfhum) [inline Core/WebAPI/Helpers/JwtTokenHelper.cs]: same here with m_

Comment (baltermia) [reply Core/WebAPI/Helpers/JwtTokenHelper.cs]: it's just a habit haha
I add a `m_` prefix for member fields and `s_` for static fields. I guess we can just use the underscore if you've used that everywhere else, I haven't really looked closer into it when writing the code.

Comment (mfhum) [reply Core/WebAPI/Helpers/JwtTokenHelper.cs]: Didn't see that being done, so I was just confused. Maybe a issue to do it, as I need to read into it.
Creating a discussion based on this

**Review by mfhum — APPROVED** [review]

**Pull request merged**